src/main/java/com/example/resourcemanagerapp/services/ResourceService.java

```diff
@@ -0,0 +1,96 @@
+package com.example.resourcemanagerapp.services;
+
+
+import com.example.resourcemanagerapp.additionalTypes.ResourceType;
+import com.example.resourcemanagerapp.dtos.AddResourceDTO;
+import com.example.resourcemanagerapp.dtos.UpdateResourceMetadataDTO;
+import com.example.resourcemanagerapp.dtos.UpdateResourceNameDTO;
+import com.example.resourcemanagerapp.exceptions.ApplicationException;
+import com.example.resourcemanagerapp.models.ResourceEntity;
+import com.example.resourcemanagerapp.models.UserEntity;
+import com.example.resourcemanagerapp.repositories.ResourceRepository;
+import com.example.resourcemanagerapp.repositories.UserRepository;
+import com.example.resourcemanagerapp.validators.ResourceApiParamsValidator;
+import com.example.resourcemanagerapp.validators.UserApiParamsValidator;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import java.time.LocalDateTime;
+
+@RequiredArgsConstructor
+@Service
+public class ResourceService {
+
+    private final ResourceRepository resourceRepository;
+    private final UserRepository userRepository;
+
+    @Transactional
+    public ResponseEntity addResource(AddResourceDTO addResourceDTO) {
```

Comment:
Jeśli używasz tych pól tylko aby wrzucić do metody validate, to może lepiej byłoby do niej przekazywać od razu ResourceDto i zrobić validateResource. Rób więcej odstępów, wtedy kod będzie czytelniejszy, np oddzielając pola od metod i przy return.